aspnetcore/hosting/url-rewriting.md

```diff
@@ -12,7 +12,293 @@ ms.technology: aspnet
+* Removing, adding, or reorganizing URL segments on incoming requests
+* Optimizing public URLs for Search Engine Optimization (SEO)
+* Permitting the use of friendly public URLs to help people predict the content they will find by following a link
+* Redirecting insecure requests to secure endpoints
+
+You can define rules for changing the URL in several ways, including regular expression (regex) matching rules, rules based on the Apache mod_rewrite module, rules based on the IIS Rewrite Module, and with your own method and class rule logic. This document introduces URL rewriting with instructions on how to use URL Rewriting Middleware in ASP.NET Core applications.
+
+> [!NOTE]
+> URL rewriting can reduce the performance of an application. Where feasible, you should design systems that don't require URL rewriting.
+
+## URL redirect and URL rewrite
+The difference in wording between *URL redirect* and *URL rewrite* may seem subtle at first but has important implications for providing resources to clients. URL Rewriting Middleware is capable of meeting the need for both.
+
+A *URL redirect* is a client-side operation, where the client is instructed to access a resource at another address. This requires a round-trip to the server, and the redirect URL returned to the client will appear in the browser's address bar when the client makes a new request for the resource. If **http://www.mysite.com/resource** is *redirected* to **http://www.mysite.com/different-resource**, the client will request **http://www.mysite.com/resource**, and the server will respond that the client should obtain the resource at **http://www.mysite.com/different-resource** with a status code indicating that the redirect is either temporary or permanent. The client will execute a new request for the resource at the redirect URL.
+
+![file.txt has been moved from folder1 into folder2 on the server. A client requests file.txt in folder1. The server sends back a 302 (Found) response with the new, temporary location of file.txt in folder2. The client makes a second request for the resource at the redirect URL. The server fetches and returns the file with a 200 (OK) status code.](url-rewriting/_static/url_redirect.png)
+
+A *URL rewrite* is a server-side operation to provide a resource from a different resource address. Rewriting a URL doesn't require a round-trip to the server, and the rewritten URL is not returned to the client and won't appear in a browser's address bar. When **http://www.mysite.com/resource** is *rewritten* to **http://www.mysite.com/different-resource**, the client will request **http://www.mysite.com/resource**, and the server will *internally* fetch the resource at **http://www.mysite.com/different-resource**. Although the client might be able to retrieve the resource at the rewritten URL, the client won't be informed that the resource exists at the rewritten URL when it makes its request and receives the response.
```

Comment:
I think it would be ok either way, but I am making the change. It was probably just the second "and" for the compound verb that made it seem long.